ni_measurementlink_service/measurement/service.py

```diff
@@ -375,14 +375,16 @@ def host_service(self) -> MeasurementService:
         """
         if self.measure_function is None:
             raise Exception("Error, must register measurement method.")
-        self.grpc_service.start(
+        port = self.grpc_service.start(
             self.measurement_info,
             self.service_info,
             self.configuration_parameter_list,
             self.output_parameter_list,
             self.measure_function,
         )
-        return self
+        if port:
+            return self
+        raise Exception("Error in starting the measurement service.")
```

Comment:
This doesn't solve the root problem, which is that DiscoveryClient does not report the error to the caller. As a result, this exception contains a generic error rather than the specific error encountered by the DiscoveryClient. It is also not linked to the original exception.

Note that I also want to pull DiscoveryClient out into a separate package, which may require fixing other issues with its interface:
- register_measurement_service uses measurement-specific types
- register_measurement_service stores the registration_id as a member variable, which assumes it is only used to register one service
- register_measurement_service function name is not symmetrical with unregister_service
  - we could deprecate register_measurement_service and add a register_service method

I don't think you should address all of that in this PR. Consider keeping the existing interface and function signatures, but change the behavior to either return True or raise an exception.

In cases where we log specific messages for specific errors, perhaps we should throw an exception with the specific message and use `raise Exception("...") from e` to link this exception to the original exception.

Reply:
@bkeryan  I have made the following changes
1) Removed except block's return False and replaced with raise Exceptions with type and message.
2) Updated to callers (`service.py`, `service_manager.py`) to catch the exception using `try except` block.
3) Updates `test_service.py`, `test_service_manager.py`, `test_discovery_client.py`.